src/Domain/LeanCode.DomainModels/Model/TimestampTz.cs

```diff
@@ -0,0 +1,47 @@
+using System.Text.Json.Serialization;
+
+namespace LeanCode.DomainModels.Model;
+
+[method: JsonConstructor]
+public sealed record class TimestampTz(DateTime UtcTimestamp, string TimeZoneId) : ValueObject
+{
+    public TimeZoneInfo GetTimeZoneInfo() => TimeZoneInfo.FindSystemTimeZoneById(TimeZoneId);
+
+    public DateTimeOffset GetLocalTimestampWithOffset() =>
+        TimeZoneInfo.ConvertTimeBySystemTimeZoneId(new DateTimeOffset(UtcTimestamp, TimeSpan.Zero), TimeZoneId);
+
+    public DateTime LocalTimestamp => TimeZoneInfo.ConvertTimeBySystemTimeZoneId(UtcTimestamp, TimeZoneId);
+
+    public DateTime UtcTimestamp { get; init; } =
+        UtcTimestamp.Kind == DateTimeKind.Utc
+            ? UtcTimestamp
+            : throw new ArgumentException("UtcTimestamp.Kind must be Utc.", nameof(UtcTimestamp));
+
+    public string TimeZoneId { get; init; } = NormalizeAndValidateTimeZoneId(TimeZoneId);
```

Comment:
At this moment I think a classic constructor would be more readable.

Reply:
But then I'd have to implement deconstructors manually :disappointed:

Reply:
I sometimes really don't know if you are sarcastic or just dead serious xD

Reply:
Nah, you guys have a point. I'll rewrite this into something where the flow should be easier to follow.

Reply:
All the boring stuff is now being done explicitly inside the constructor.